### PR TITLE
lbrtc OK

### PR DESCRIPTION
**Pull request by Prakxo**
Partially documented, has made-up names according to their usages in `lb_rtc` if they can be inferred
Open to name change/s though
It's not grouped, but open for suggestions if they needed to be grouped or not

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -245,6 +245,22 @@ build/src/boot/libu64/%.o: OPTFLAGS := -O2
...
+build/src/code/lbrtc_init.o: CC = $(CC_OLD)
+build/src/code/lbrtc_init.o: OPTFLAGS = -O1 -g1
+build/src/code/7220C0.o: CC = $(CC_OLD)
+build/src/code/7220C0.o: OPTFLAGS = -O1 -g1
+build/src/code/722570.o: CC = $(CC_OLD)
+build/src/code/722570.o: OPTFLAGS = -O1 -g1
+build/src/code/722880.o: CC = $(CC_OLD)
+build/src/code/722880.o: OPTFLAGS = -O1 -g1
+build/src/code/722AB0.o: CC = $(CC_OLD)
+build/src/code/722AB0.o: OPTFLAGS = -O1 -g1
```
Comment (AngheloAlf) [inline Makefile:261]: I think we could move those into a `lbrtc` folder inside `code`. This way we could simplify these rules by applying them to the whole directory

File: src/code/722AB0.c [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                                        return 0;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
```
Comment (AngheloAlf) [inline src/code/722AB0.c:112]: Could you remove the `else`s that follow a `return`?

File: include/lbrtc.h
Comment (AngheloAlf) [inline include/lbrtc.h:1]: I'm a bit confused by the existence of both `lb_rtc.h` and `lbrtc.h`. Why both?

File: src/code/lb_rtc.c [hunk 3]
```diff
@@ -97,7 +92,7 @@ void lbRTC_SetTime(lbRTC_time_c* time) {
     if (common_data.time.rtcEnabled == 1 && !common_data.time.rtcCrashed) {
         time->weekday = lbRTC_Week(time->year, time->month, time->day);
         mq = padmgr_LockSerialMesgQ();
-        func_800FE670_jp(mq, time);
+        lbrtc_setTime(mq, (OSRTCTime*)time);
```
Comment (AngheloAlf) [inline src/code/lb_rtc.c:95]: Looks like `lbRTC_time_c` and `OSRTCTime` are the same struct. Could you merge them?

File: include/lbrtc.h
Comment (Prakxo) [reply include/lbrtc.h:1]: That's what I named this back in the day when I started working on it, `lbrtc` is the libultra internal header and `lb_rtc` is completely specific to the game, how should I name the libultra header if `lbrtc.h` is confusing?

File: src/code/lb_rtc.c (hunk 3 above)
Comment (Prakxo) [reply src/code/lb_rtc.c:95]: The existence of these two structs are known by the flashrom structs in the symbols, should I just have `lbRTC_time_c` have an `OSRTCTime` member?

File: src/code/722AB0.c (hunk 2 above)
Comment (Prakxo) [reply src/code/722AB0.c:112]: Doesn't match
https://decomp.me/scratch/o78Jn

Comment (AngheloAlf) [reply src/code/722AB0.c:112]: In that case, could you merge the `else`s with the `if`s? It reduces the indentation a lot.
Like this: https://decomp.me/scratch/W9AHv

File: include/lbrtc.h
Comment (AngheloAlf) [reply include/lbrtc.h:1]: There's no real time clock support on libultra to my knowledge.
From a quick look it seems like `lbrtc` is an external library that the original developers used (like `libultra`) and `lb_rtc` is a game file that wraps the logic around the library.
I guess its fine to have both.

File: src/code/lb_rtc.c (hunk 3 above)
Comment (AngheloAlf) [reply src/code/lb_rtc.c:95]: Could you point me to those symbols? I want to take a look

**Pull request merged**